day_08.md

```diff
@@ -0,0 +1,188 @@
+# [A walk in JavaScript](/README.md)
+
+## DAY 8
+
+- Classes
+  - General definition
+  - Syntax
+    - `class` declaration statement
+    - `class` expression
+    - Class body and method definitions
+  - ES6 Classes in depth
+- OOP vs Functional
+  - General definitions
+  - Some essential differences
+  - Examples
+- Exercises
+
+## Classes
+
+### General definition
+
+Let's start from the definition:
+
+> JavaScript classes, introduced in ECMAScript 2015, are primarily **syntactical sugar** over JavaScript's existing **prototype-based** inheritance. The class **syntax does not introduce a new object-oriented inheritance model** to JavaScript.
+>
+> Source: [MDN - Classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes)
+
+(－‸ლ) ok, so, the name is  `class` but it seems they're not classes. Let's continue.
+
+> Classes are in fact "special [functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions)", and just as you can define [function expressions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/function) and [function declarations](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function), the class syntax has two components: [class expressions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/class) and [class declarations](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/class).
+>
+> Source: [MDN - Classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes)
+
+(－‸ლ) (－‸ლ) now I'm confused ...so ... they're not  [classes](https://en.wikipedia.org/wiki/Class_(computer_programming)) so technically it's NOT a [class-based](https://en.wikipedia.org/wiki/Class-based_programming) language BUT you CAN do [OOP](https://en.wikipedia.org/wiki/Object-oriented_programming)  ... so what in the world are they and what's the use? 
+
+Please don't give up. Despite the "unhappy" and misleading naming, they are a very useful and powerful abstraction around repetirtive and error prone routines to create `prototype` based inheritance structures.
+
+This time, ECMAScript international isn't very friendly with this:
+
+- [Class Definitions](https://www.ecma-international.org/ecma-262/6.0/#sec-class-definitions)
+- [Functions and Classes](https://www.ecma-international.org/ecma-262/6.0/#sec-functions-and-classes)
+- [MakeClassConstructor](https://www.ecma-international.org/ecma-262/6.0/#sec-makeclassconstructor)
+
+So let's borrough some info from MDN.
```

Comment:
Typo fix => borrough should be borough

Reply:
> Typo fix => borrough should be borough

I think he meant borrow.
https://www.macmillandictionary.com/dictionary/british/borough
https://www.macmillandictionary.com/dictionary/british/borrow

Reply:
Yes, @luispagliaro you're right, so
Typo fix => borrough should be borrow